### PR TITLE
Add javaFx for buildBoard

### PR DESCRIPTION
**Pull request by patiem**

### CODE REVIEW

File: client/src/main/java/building/FleetDrop.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    private final double SCENE_WIDTH = 800;
+    private final double SCENE_HEIGHT = 600;
+    private final String START_BOARD_URL = "/fxmls/buildBoardAllShips.fxml";
+
+    @Override
+    public void start(Stage stage) throws Exception {
+        stage.setTitle("Build Board");
+        stage.setResizable(false);
+
+        FXMLLoader loader = new FXMLLoader(getClass().getResource(START_BOARD_URL));
```
Comment (M-Aarsman) [inline client/src/main/java/building/FleetDrop.java]: can be final

File: client/src/main/java/building/FleetDrop.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        AnchorPane playBoard = loader.load();
+
+        Scene scene = new Scene(playBoard, SCENE_WIDTH, SCENE_HEIGHT);
```
Comment (M-Aarsman) [inline client/src/main/java/building/FleetDrop.java]: can be final

File: client/src/main/java/building/FleetDropController.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    private static final int GRID_SIZE = 10;
+
+    @FXML
+    GridPane shipBoard;
+
+    @FXML
+    public Pane port;
+
+    @FXML
+    public Rectangle ship4, ship3, ship3a, ship2, ship2a, ship2b, ship1, ship1a, ship1b, ship1c;
```
Comment (M-Aarsman) [inline client/src/main/java/building/FleetDropController.java]: for me it is unreadable, when more then 1 variable is initialize int the 1 line http://www.informit.com/articles/article.aspx?p=2133373&seqNum=2

File: client/src/main/java/building/FleetDropController.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    private Rectangle buildShip;
+
+    Fleet fleet;
```
Comment (M-Aarsman) [inline client/src/main/java/building/FleetDropController.java]: why not private?

File: client/src/main/java/building/FleetDropController.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    Sea sea = new Sea();
```
Comment (M-Aarsman) [inline client/src/main/java/building/FleetDropController.java]: the same

File: client/src/main/java/building/FleetDropController.java [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+                field.reset();
+                event.consume();
+            };
+
+    EventHandler<MouseEvent> dragDetected =
+            event -> {
+                buildShip = (Rectangle) event.getSource();
+                Dragboard db = buildShip.startDragAndDrop(TransferMode.COPY);
+                ClipboardContent content = new ClipboardContent();
+                content.putString("Hello!");
```
Comment (M-Aarsman) [inline client/src/main/java/building/FleetDropController.java]: it some kind of test string?

File: client/src/main/java/model/Fleet.java [hunk 7]
```diff
@@ -8,38 +8,39 @@
...
-    private void addFirstMastToShip(Mast mast) {
+    private void addFirstMastToFleet(Mast mast) {
         masts.add(mast);
     }
 
-    public void startToBuilOneShip(Mast mast) {
-        addFirstMastToShip(mast);
-        Ship ship = new Ship(mast);
-        lastShip = ship;
+    public void startToBuildOneShip(Mast mast, int shipLength) {
```
Comment (M-Aarsman) [inline client/src/main/java/model/Fleet.java]: only start build or it build? :)

File: client/src/main/java/model/Ship.java [hunk 8]
```diff
@@ -10,13 +10,13 @@
 
     private static final int START_SIZE = 1;
 
-    private final Mast[] masts;
+    private Mast[] masts;
```
Comment (M-Aarsman) [inline client/src/main/java/model/Ship.java]: cant be final?

File: client/src/main/java/model/Ship.java [hunk 9]
```diff
@@ -10,13 +10,13 @@
...
     private final int length;
     private int buildLength;
```
Comment (M-Aarsman) [inline client/src/main/java/model/Ship.java]: can't be final?

File: client/src/test/java/model/BoundariesPositionTest.java [hunk 10]
```diff
@@ -0,0 +1,7 @@
+package model;
+
+import static org.testng.Assert.*;
+
+public class BoundariesPositionTest {
```
Comment (M-Aarsman) [inline client/src/test/java/model/BoundariesPositionTest.java]: do not push empty class

File: client/src/main/java/model/Ship.java (hunk 9 above)
Comment (patiem) [reply client/src/main/java/model/Ship.java]: No, this is increased during building.

File: client/src/test/java/model/BoundariesPositionTest.java (hunk 10 above)
Comment (patiem) [reply client/src/test/java/model/BoundariesPositionTest.java]: I dont know why it has been pushed :/

File: client/src/main/java/model/Ship.java (hunk 9 above)
Comment (M-Aarsman) [reply client/src/main/java/model/Ship.java]: ok

File: client/src/test/java/model/BoundariesPositionTest.java (hunk 10 above)
Comment (M-Aarsman) [reply client/src/test/java/model/BoundariesPositionTest.java]: ok

File: client/src/main/java/building/FleetDropController.java (hunk 6 above)
Comment (patiem) [reply client/src/main/java/building/FleetDropController.java]: I change this string to "Ship", I'm not sure why, but this string must be set for content to work.

File: client/src/main/java/model/Fleet.java (hunk 7 above)
Comment (patiem) [reply client/src/main/java/model/Fleet.java]: Only start , rest of building is somewhere else

File: client/src/main/java/model/Ship.java (hunk 9 above)
Comment (patiem) [reply client/src/main/java/model/Ship.java]: No buildLength is changing during bulding ship

File: client/src/test/java/model/BoundariesPositionTest.java (hunk 10 above)
Comment (patiem) [reply client/src/test/java/model/BoundariesPositionTest.java]: It shouldnt be pushed :/ I add one test to fill it with something

**Pull request merged**